components/style/attr.rs

```diff
@@ -603,3 +603,89 @@ pub struct AttrIdentifier {
     pub namespace: Namespace,
     pub prefix: Option<Prefix>,
 }
+
+/// Valid date string should be yyyy-mm-dd
+/// https://html.spec.whatwg.org/multipage/#valid-date-string
+pub fn is_valid_date_string(value: &str) -> bool {
+    // step 2
+    let value_vec: Vec<&str> = value.split('-').collect();
+    if !(value_vec.len() == 3) {
+        return false;
+    }
+    let year = value_vec[0];
+    let month = value_vec[1];
+    let day = value_vec[2];
+
+    // setp 1
+    let month_string = format!("{}-{}", year, month);
```

Comment:
This is pretty awkward, we've just split the string on `-`, but here we're heap-allocating a String with a `-` to parse a month string.

Reply:
`YYYY-MM-DD`

I want to get `YYYY-MM`. I don't know how to write it prettier.